python/TestHarness/testers/RunApp.py

```diff
@@ -108,6 +108,9 @@ def getCommand(self, options):
       # also, neither is the conflicting option "warn-unused"
       specs['cli_args'].append('--error-unused')
 
+    if '--check-input' in specs['cli_args']:
```

Comment:
I'd prefer that we make this an actual parameter instead (e.g. `params.addParam(`check-input`, False, ...`). If that parameter is set we'll add the right `cli_args` option for the user and change the status_message. For that matter, we really should create a new method on Tester to return that status message instead of communicating back through the input parameters to the TestHarness.